torch/csrc/autograd/autograd.cpp

```diff
@@ -92,6 +93,11 @@ variable_list run_backward(
       if (!grad_fn) {
         grad_fn = impl::try_get_grad_accumulator(input);
       }
+      if (accumulate_grad) {
+        TORCH_CHECK(
+          input.is_leaf(),
+          "One of the differentiated Tensors is not a leaf Tensor");
```

Comment:
nit: I would improve this error to specify that this differentiated Tensor is one of the "inputs" given to `.backward()`.